core/src/banking_stage/transaction_scheduler/scheduler_controller.rs

```diff
@@ -316,25 +429,36 @@ impl SchedulerController {
         let mut error_counts = TransactionErrorMetrics::default();
         for chunk in packets.chunks(CHUNK_SIZE) {
             let mut post_sanitization_count: usize = 0;
-            let (transactions, fee_budget_limits_vec): (Vec<_>, Vec<_>) = chunk
+
+            let mut arc_packets = Vec::with_capacity(chunk.len());
+            let mut transactions = Vec::with_capacity(chunk.len());
+            let mut fee_budget_limits_vec = Vec::with_capacity(chunk.len());
+
+            chunk
                 .iter()
                 .filter_map(|packet| {
-                    packet.build_sanitized_transaction(feature_set, vote_only, bank.as_ref())
+                    packet
+                        .build_sanitized_transaction(feature_set, vote_only, bank.as_ref())
+                        .map(|tx| (packet.clone(), tx))
                 })
                 .inspect(|_| saturating_add_assign!(post_sanitization_count, 1))
-                .filter(|tx| {
+                .filter(|(_packet, tx)| {
                     SanitizedTransaction::validate_account_locks(
                         tx.message(),
                         transaction_account_lock_limit,
                     )
                     .is_ok()
                 })
-                .filter_map(|tx| {
+                .filter_map(|(packet, tx)| {
                     process_compute_budget_instructions(tx.message().program_instructions_iter())
-                        .map(|compute_budget| (tx, compute_budget.into()))
+                        .map(|compute_budget| (packet, tx, compute_budget.into()))
                         .ok()
                 })
-                .unzip();
```

Comment:
`itertools` _does_ have an unzip that emits more than a 2-tuple, but this is probably actually better (with previous comment)